snforge_std/src/signing/stark_curve.cairo

```diff
@@ -0,0 +1,56 @@
+use starknet::testing::cheatcode;
+use ecdsa::check_ecdsa_signature;
+
+#[derive(Copy, Drop)]
+struct StarkCurveKeyPair {
+    private_key: felt252,
+    public_key: felt252
+}
+
+trait StarkCurveKeyPairTrait {
```

Comment:
Traits should abstract over properties, not be struct specific. Also please make it consistent with rust `signature` crate.
Please split it into `Signer` `Verifier`. `generate` and `from_private_key` shouldn't be part of the trait.